checkasm/meson.build

```diff
@@ -0,0 +1,33 @@
+checkasm_src = files(
+    'checkasm.c',
+    'rasterizer.c',
+    'blend_bitmaps.c',
+    'be_blur.c',
+    'blur.c',
+)
+
+checkasm_src_x86 = files(
+    'x86/checkasm.asm',
+)
+checkasm_src_aarch64 = files(
+    'arm/checkasm_64.S',
+)
+
+checkasm_asm_sources = []
+if generic_cpu_family == 'x86'
+    checkasm_src += checkasm_src_x86
+elif generic_cpu_family == 'aarch64'
+    checkasm_src += checkasm_src_aarch64
+endif
+
+libass_checkasm = executable(
+    'checkasm',
+    checkasm_src + config_h,
+    install: false,
+    include_directories: incs,
+    dependencies: deps,
+    link_with: libass_for_tools,
```

Comment:
incorrect linkage; this will break on dynamic-only builds on Windows since checkasm relies on accessing private libass functions *(only on Windows because meson still lacks cross-platform symbol export control)*

Reply:
Should I just disable checkasm and compare when building shared-only? (`compare` is already off by default)

Reply:
In autotools we split the actual libass lib build into two stages *(building internal static-only lib; pushing this through the linker to get the final lib)* to be able to always provide a static lib for checkasm, so if you want to keep `ass_filesystem` in `compare`, you can use this too.
Given meson’s position as a secondary build system with limited support, just erroring out if a tool needing internal API is requested in a dynamic-only build is probably fine too.